programs/drift/src/math/spot_withdraw.rs

```diff
@@ -36,13 +35,24 @@ pub fn calculate_max_borrow_token(
 
 pub fn check_user_exception_to_withdraw_limits(
     spot_market: &SpotMarket,
-    spot_position: Option<&SpotPosition>,
+    user: Option<&User>,
     token_amount_withdrawn: Option<u128>,
 ) -> DriftResult<bool> {
     // allow a smaller user in QUOTE_SPOT_MARKET_INDEX to bypass and withdraw their principal
     let mut valid_user_withdraw = false;
-    if let Some(spot_position) = spot_position {
-        if spot_position.market_index == QUOTE_SPOT_MARKET_INDEX
+    if let Some(user) = user {
+        let spot_position = user.get_spot_position(spot_market.market_index).unwrap();
+        let net_deposits = user
```

Comment:
do you not need to compare this to the withdraw amount because total deposits/withdraws have already been updated?

Reply:
changed order for total withdraw update